TwipsterApp/LoginWindow.xaml.cs

```diff
@@ -7,15 +11,41 @@ namespace TwipsterApp
     /// </summary>
     public partial class LoginWindow : Window
     {
+        public static User currentUser;
         public LoginWindow()
         {
             InitializeComponent();
         }
 
         private void RegisterButton_Click(object sender, RoutedEventArgs e)
         {
-            RegistrationWindow registrationWindow = new RegistrationWindow();
+            var registrationWindow = new RegistrationWindow();
+
             registrationWindow.Show();
         }
+
+        private void OnLoginButtonClicked(object sender, RoutedEventArgs e)
+        {
+            //var user = new User();
+            var twipsterMainWindow = new TwipsterMainWindow();
+
+            var validators = new List<IValidator>
+            {
+                new PasswordValidator()
+            };
+
+            try {
+                using (var context = new TwipsterDbContext())
+                {
+                    currentUser = context.Users.Single(x => x.Login == LoginTexBox.Text);
+                }
+                validators[0].Validate(currentUser, PasswordTexBox.Text);
+
+                twipsterMainWindow.Show();
```

Comment:
Show and Close should be called after try catch statements. It should not throw exception in expected way.